### DISCUSSION TITLE
Warning if there is a TODO comment in the AL code

### DISCUSSION THREAD

**Discussion by rvanbekkum** [StefanMaron/BusinessCentral.LinterCop#86] [New Rule]
Currently we have implemented a step in our build pipelines that will warn about TODO comments in the repository, but it's currently simply a PowerShell task that goes through all files in the repository and checks for TODO.
This could be nice to have as a code analyzer rule so you also see this in VS Code (and apart from that, I think it could be useful for others as well to be warned that there are TODOs somewhere in their projects).

**Comment by fvet** [comment]
@rvanbekkum We apply a zero-warning policy for builds to succeed. For AL0432 that show up, we apply following logic:

- When it's a AL0432 warning raised by ourselves / our ISV objects (because we obsoleted some stuff - but we cannot delete yet the reference to it) we add a `#pragma warning disable AL0432`. The warning has been reviewed and taken care of, so we disable the warning
(PS: we don't use the #CLEAN pre-compiler directive pattern)

- When it's a AL0432 warning raised by MS / BC objects (but we can't solve the warning yet in the current release) we add a `#pragma warning disable AL0432 // TODO - Review in BC20`. The warning has been temporarily reviewed (for the pipeline not to fail), but is still marked as to review with the next major.

In this case, the // TODO tag make sense to us. How do you deal with this scenario?

**Reply by rvanbekkum** [reply]
More or less the same. 😊👍

**Comment by StefanMaron** [comment]
I had a similar idea recently! But maybe TODO is not the right one. I feel like it would just be nice to be able to mark a line in code to manually create a warning. This will then fail the build of you forgot to remove that line. Just like a reminder to not publish code which is not meant to be released.

**Comment by StefanMaron** [comment]
How about `//TODO`/`// TODO` by default with option to change this via `LinterCop.json` ?
Then it could be changed to something like `// TEMP` 
Obviously the setting would be a list of strings to warn.

I would tend to set this rule to disabled by default

**Reply by rvanbekkum** [reply]
Yes that would be nice. And I agree with setting it disabled by default.
